### PR TITLE
Session management

### PR DESCRIPTION
**Pull request by 9gustin**
## Summary

Add session managment with local storage.
Add session validation on home page.
Add logout to remove session.

## Screenshots

image

## Trello Card

[Session management & Logout](https://trello.com/c/DAEvP3eE/75-session-management-logout)

### CODE REVIEW

File: react/w-books/src/app/screens/Home/index.tsx [hunk 1]
```diff
@@ -1,9 +1,17 @@
 import React from 'react'
+import { Redirect } from 'react-router';
+import Navbar from '~components/Navbar'
+import PATHS from '~components/Routes/paths';
+
+import useSession from '../../../hooks/useSession';
 
 export default function Home() {
```
Comment (LucasZibell) [inline react/w-books/src/app/screens/Home/index.tsx:8]: export at the end of the file

File: react/w-books/src/hooks/useSession.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { deleteData, getData, saveData, SESSION } from "../utils/manageData";
```
Comment (LucasZibell) [inline react/w-books/src/hooks/useSession.ts:1]: Use single quotes

File: react/w-books/src/hooks/useSession.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  const startSession = (accessToken:string) => {
+    saveData(SESSION, accessToken);
+  };
+
+  const getSession = () => {
+    const session = getData(SESSION);
+
+    return session;
+  };
```
Comment (LucasZibell) [inline react/w-books/src/hooks/useSession.ts:20]:
```
  const getSession = () => getData(SESSION);
```

File: react/w-books/src/hooks/useSession.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import { deleteData, getData, saveData, SESSION } from "../utils/manageData";
+
+const useSession = () => {
+
+  const isLogged = (): boolean => {
+    const session = getSession();
+    if (session) return true;
+
+    return false;
+  };
```
Comment (LucasZibell) [inline react/w-books/src/hooks/useSession.ts:10]:
```
  const isLogged = (): boolean => getSession();
```

File: react/w-books/src/hooks/useSession.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  const getSession = () => {
+    const session = getData(SESSION);
+
+    return session;
+  };
+
+
+  const stopSession = () => {
+    deleteData(SESSION);
+  };
```
Comment (LucasZibell) [inline react/w-books/src/hooks/useSession.ts:25]:
```
  const stopSession = () => deleteData(SESSION);
```

File: react/w-books/src/hooks/useSession.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+  return {
+    isLogged,
+    startSession,
+    stopSession,
+  };
+};
+
+export default useSession;
```
Comment (LucasZibell) [inline react/w-books/src/hooks/useSession.ts:34]: What logic are you abstracting here... seems you are just renaming functions

File: react/w-books/src/services/userService.ts [hunk 7]
```diff
@@ -1,16 +1,22 @@
...
-  
-  if(response.ok){
+
+  if (response.ok) {
     const headers:any = response.headers;
-    const accessToken = headers['access-token'];
-    console.log(accessToken);
+    const token = headers['access-token'];
+
+    handleAccessToken?.(token);
```
Comment (LucasZibell) [inline react/w-books/src/services/userService.ts:19]: Check previous PR comments

File: react/w-books/src/utils/manageData.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
+export const SESSION = 'SESSION';
+
+const saveData = (key: typeof SESSION, value: any) => {
+  if (typeof localStorage !== 'undefined') {
```
Comment (LucasZibell) [inline react/w-books/src/utils/manageData.ts:4]: When will the LocalStorage be undefined?

File: react/w-books/src/utils/manageData.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    const data = JSON.stringify(value);
+    localStorage.setItem(key, data);
+  }
+};
+
+const getData = (key: typeof SESSION) => {
+  if (typeof localStorage !== 'undefined') {
```
Comment (LucasZibell) [inline react/w-books/src/utils/manageData.ts:11]: same

File: react/w-books/src/utils/manageData.ts [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+    if (data) {
+      return JSON.parse(data);
+    }
+  }
+  return null;
+};
+
+const deleteData = (key: typeof SESSION) => {
+  if (typeof localStorage !== 'undefined') {
```
Comment (LucasZibell) [inline react/w-books/src/utils/manageData.ts:22]: same

File: react/w-books/src/utils/manageData.ts [hunk 11]
```diff
@@ -0,0 +1,27 @@
+export const SESSION = 'SESSION';
+
+const saveData = (key: typeof SESSION, value: any) => {
```
Comment (LucasZibell) [inline react/w-books/src/utils/manageData.ts:3]: export const

File: react/w-books/src/utils/manageData.ts [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+  if (typeof localStorage !== 'undefined') {
+    const data = JSON.stringify(value);
+    localStorage.setItem(key, data);
+  }
+};
+
+const getData = (key: typeof SESSION) => {
```
Comment (LucasZibell) [inline react/w-books/src/utils/manageData.ts:10]: export const

File: react/w-books/src/utils/manageData.ts [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+    let data = localStorage.getItem(key);
+
+    if (data) {
+      return JSON.parse(data);
+    }
+  }
+  return null;
+};
+
+const deleteData = (key: typeof SESSION) => {
```
Comment (LucasZibell) [inline react/w-books/src/utils/manageData.ts:21]: export const

**Review by LucasZibell — CHANGES_REQUESTED** [review]

**Comment by 9gustin** [comment]
changes are ready for review

**Review by LucasZibell — DISMISSED** [review]

**Review by DavidRojas1612 — DISMISSED** [review]

**Review by DavidRojas1612 — APPROVED** [review]

**Pull request merged**